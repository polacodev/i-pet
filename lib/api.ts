import { decode } from 'base64-arraybuffer';

import { supabase } from '@/lib/supabase';
import { PetProps } from '@/types/pet';
import { Session } from '@/types/session';
import { UserProfileProps } from '@/types/user';

export const getPetList = async (userId: string | null) => {
  let petList: PetProps[] = [];
  const { data, error } = await supabase
    .from('pets')
    .select(
      `
      id,
      inserted_at,
      pet_image,
      pet_age,
      pet_breed,
      pet_gender,
      pet_medical_condition,
      pet_name,
      pet_type,
      user_id,
      profiles (
        id,
        email,
        country_code,
        phone
      )
      `,
    )
    .eq('user_id', userId);

  // if (error) return { data, error };
  // if (data) {
  petList = data;
  return { data: petList, error };
  // }
  // return { data, error };
};

export const getPetById = async (petId: string) => {
  const { data, error } = await supabase
    .from<any, 'public'>('pets')
    .select(
      `
      id,
      inserted_at,
      pet_image,
      pet_age,
      pet_breed,
      pet_gender,
      pet_medical_condition,
      pet_name,
      pet_type,
      user_id,
      profiles (
        id,
        email,
        country_code,
        phone
      )
    `,
    )
    .eq('id', petId)
    .single();
  return { data, error };
};

export const deletePetById = async (petId: string) => {
  const { error } = await supabase.from('pets').delete().eq('id', petId);
  return { error };
};

export const insertPet = async (newPet: any) => {
  const { error } = await supabase.from('pets').upsert(newPet);
  return { error };
};

export const storePetImage = async (filePath: string, code: any, contentType: string) => {
  const { data, error } = await supabase.storage
    .from('pets')
    .upload(filePath, decode(code), { contentType });
  return { data, error };
};

export const deletePetImage = async (petImage: string) => {
  const { data, error } = await supabase.storage.from('pets').remove([petImage]);
  return { data, error };
};

export const signInWithPassword = async (email: string, password: string) => {
  const { data, error } = await supabase.auth.signInWithPassword({
    email,
    password,
  });
  return { data, error };
};

export const signUpWithEmailPassword = async (email: string, password: string) => {
  const { data, error } = await supabase.auth.signUp({ email, password });
  return { data, error };
};

export const saveUserProfile = async (user: UserProfileProps) => {
  const { data, error } = await supabase.from('profiles').upsert(user);
  return { data, error };
};

export const signOut = async () => {
  await supabase.auth.signOut();
};

export const getSession = async () => {
  return await supabase.auth.getSession();
};

export const onAuthStateChange = (setSession: (session: Session | null) => void) => {
  return supabase.auth.onAuthStateChange((_event, session) => {
    setSession(session);
  });
};
