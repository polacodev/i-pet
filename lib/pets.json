[
  {
    "id": "550e8400-e29b-41d4-a716-446655440000",
    "petName": "Bella",
    "petType": "Dog",
    "petAge": 3,
    "petBreed": "Golden Retriever",
    "petGender": "Female",
    "petMedicalCondition": "Hip Dysplasia",
    "ownerName": "John Smith",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440001",
    "petName": "Max",
    "petType": "Dog",
    "petAge": 2,
    "petBreed": "Bulldog",
    "petGender": "Male",
    "petMedicalCondition": "Allergies",
    "ownerName": "Emily Johnson",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440002",
    "petName": "Luna",
    "petType": "Cat",
    "petAge": 1,
    "petBreed": "Siamese",
    "petGender": "Female",
    "petMedicalCondition": "Asthma",
    "ownerName": "Michael Brown",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440003",
    "petName": "Charlie",
    "petType": "Dog",
    "petAge": 5,
    "petBreed": "Labrador",
    "petGender": "Male",
    "petMedicalCondition": "Arthritis",
    "ownerName": "Jessica Davis",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440004",
    "petName": "Daisy",
    "petType": "Dog",
    "petAge": 4,
    "petBreed": "Beagle",
    "petGender": "Female",
    "petMedicalCondition": "Diabetes",
    "ownerName": "David Wilson",
    "ownerPhone": "60717546"
  }
]