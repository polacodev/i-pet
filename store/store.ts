import { create } from 'zustand';

import { PetProps } from '@/types/pet.type';

type PetStore = {
  pets: PetProps[];
  setPetList: (petList: PetProps[]) => void;
  removePetFromStore: (petId: string) => void;
};

export const usePetStore = create<PetStore>((set) => ({
  pets: [],
  setPetList: (petList: PetProps[]) => set({ pets: petList }),
  removePetFromStore: (petId: string) =>
    set((state) => ({ pets: state.pets.filter((pet) => pet.id !== petId) })),
}));
