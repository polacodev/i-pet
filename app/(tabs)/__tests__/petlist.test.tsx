import { jest, test } from '@jest/globals';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import { router } from 'expo-router';
import '@testing-library/react-native/extend-expect';
import React, { useState as useStateMock } from 'react';
import { Alert, RefreshControl } from 'react-native';

import PetList from '@/app/(tabs)/petlist';

const petObj = {
  pet: {
    id: 'testId',
    inserted_at: 'testInsertedId',
    pet_image: 'https://fakeurl/fakeimage/test.png',
    pet_age: 'testPetAge',
    pet_breed: 'testPetBreed',
    pet_gender: 'testPetGender',
    pet_medical_condition: 'testPetMC',
    pet_name: 'testPetName',
    pet_type: 'testPetType',
    profiles: {
      country_code: 'testProfileCountryCode',
      email: 'testProfileEmail',
      id: 'testProfileId',
      phone: 'testProfilePhone',
    },
    user_id: 'testUserId',
  },
};

const petObjNoImage = {
  pet: {
    id: 'testId',
    inserted_at: 'testInsertedId',
    pet_image: null,
    pet_age: 'testPetAge',
    pet_breed: 'testPetBreed',
    pet_gender: 'testPetGender',
    pet_medical_condition: 'testPetMC',
    pet_name: 'testPetName',
    pet_type: 'testPetType',
    profiles: {
      country_code: 'testProfileCountryCode',
      email: 'testProfileEmail',
      id: 'testProfileId',
      phone: 'testProfilePhone',
    },
    user_id: 'testUserId',
  },
};

jest.mock('@/lib/supabase', () => ({
  supabase: {
    auth: {
      signInWithPassword: jest.fn(),
    },
    from: jest.fn(() => ({
      select: jest.fn().mockReturnThis(),
      insert: jest.fn().mockReturnThis(),
      eq: jest.fn().mockReturnThis(),
      single: jest.fn().mockReturnThis(),
    })),
  },
}));

jest.mock('@/lib/api', () => ({
  getPetList: jest.fn(),
  deletePetById: jest.fn(),
  deletePetImage: jest.fn(),
}));

const setState = jest.fn();

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

jest.mock('@/store/store', () => ({
  usePetStore: jest.fn(),
  useToastStore: jest.fn(),
}));

jest.mock('@/components/context/UserContext', () => ({
  useUser: jest.fn(),
}));

jest.mock('expo-router', () => ({
  router: {
    navigate: jest.fn(),
    push: jest.fn(),
  },
}));

jest.mock('@react-native-async-storage/async-storage', () => ({
  AsyncStorage: jest.fn(),
}));

jest.mock('@/localizations/localization', () => ({
  localization: {
    t: jest.fn().mockImplementation((key) => key),
  },
}));

describe('<PetList />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
    useStateMock.mockImplementation((init: boolean) => [init, setState]);
    (router.navigate as jest.Mock).mockClear();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
    jest.resetAllMocks();
  });

  test('PetList with No User Session renders correctly and Snapshot', () => {
    useStateMock.mockImplementationOnce((init: boolean) => [init, setState]);
    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: [],
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: null,
    });

    const { getByText, toJSON } = render(<PetList />);
    expect(getByText('pet_list_message_3')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  test('PetList Data empty with User Session renders correctly and Snapshot', () => {
    useStateMock.mockImplementationOnce(() => [true, setState]);

    require('@/lib/api').getPetList.mockReturnValue({
      data: [],
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: [],
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { toJSON } = render(<PetList />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('navigate to /register event is being called when add-circle button is pressed', () => {
    useStateMock.mockImplementationOnce(() => [true, setState]);

    require('@/lib/api').getPetList.mockReturnValue({
      data: [],
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: [],
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
      user: { id: 'user-id1' },
    });

    const { getByTestId } = render(<PetList />);

    act(() => {
      fireEvent.press(getByTestId('pet-button-register'), 'onClick');
    });

    expect(router.navigate).toHaveBeenCalledWith('/register');
  });

  test('setLoading true makes appear the ActivityIndicator and Snapshot', () => {
    useStateMock.mockImplementationOnce(() => [true, setState]);
    useStateMock.mockImplementationOnce(() => [true, setState]);

    require('@/lib/api').getPetList.mockReturnValue({
      data: [],
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: [],
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { getByText, toJSON } = render(<PetList />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('goToPetRegister is called with User Session and PetList and Snapshot', () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { getByTestId, toJSON } = render(<PetList />);

    act(() => {
      fireEvent.press(getByTestId('pet-button-register-with-empty-data'), 'onClick');
    });

    expect(getByTestId('pet-button-register-with-empty-data')).toBeTruthy();
    expect(router.navigate).toHaveBeenCalledWith('/register');
    expect(toJSON()).toMatchSnapshot();
  });

  test('goToPetDetails is called when Pet Card is pressed', () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { getByTestId } = render(<PetList />);

    act(() => {
      fireEvent.press(getByTestId('pet-details-card'), 'onClick');
    });

    expect(router.push).toHaveBeenCalledWith({
      pathname: '/details/testId',
      params: { petData: JSON.stringify(Object.values(petObj)[0]) },
    });
  });

  test('PetCard does not contain a Pet Image instead of it renders an aux icon', () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObjNoImage),
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObjNoImage),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { getByTestId } = render(<PetList />);

    expect(getByTestId('pet-icon-aux')).toBeTruthy();
  });

  test('refreshingPetList is called then refreshing event is enabled', async () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: jest.fn(),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const { UNSAFE_getByType } = render(<PetList />);
    const refreshControl = UNSAFE_getByType(RefreshControl);
    expect(refreshControl.props.refreshing).toBe(false);

    fireEvent(refreshControl, 'refresh');

    await waitFor(() => expect(refreshControl.props.refreshing).toBe(false));
  });

  test('onDeletePetById event and Alert (Dismiss, Ok) events are called when delete card icon is pressed', async () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: jest.fn(),
    });

    require('@/lib/api').deletePetById.mockReturnValue({
      data: null,
      error: new Error('deletePetById test error'),
    });

    require('@/lib/api').deletePetImage.mockReturnValue({
      data: null,
      error: new Error('deletePetById test error'),
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const deleteAlert = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<PetList />);

    const onOKPress = jest.spyOn(Alert, 'alert');

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-delete-card-icon'), 'onClick');
    });
    expect(deleteAlert).toHaveBeenCalled();

    const onOK = deleteAlert.mock.calls[0]?.[2]?.[1];
    const onCancel = deleteAlert.mock.calls[0]?.[2]?.[0];
    onOK?.onPress();
    onCancel?.onPress();

    expect(onOKPress).toHaveBeenCalled();
  });

  test('deletePetById, deletePetImage are called when Alert Ok event is pressed', async () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: null,
    });

    require('@/lib/api').deletePetById.mockReturnValue({
      data: jest.fn(),
      error: null,
    });

    require('@/lib/api').deletePetImage.mockReturnValue({
      data: jest.fn(),
      error: null,
    });

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const deleteAlert = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<PetList />);

    const onOKPress = jest.spyOn(Alert, 'alert');

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-delete-card-icon'), 'onClick');
    });
    expect(deleteAlert).toHaveBeenCalled();

    const onOK = deleteAlert.mock.calls[0]?.[2]?.[1];
    onOK?.onPress();

    expect(onOKPress).toHaveBeenCalled();
  });

  test('deletePetById throws an error that is not an instance of ERROR ', async () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: null,
    });

    // deletePetById catch and ERROR is not instance of ERROR
    require('@/lib/api').deletePetById.mockRejectedValue(123);

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
      user: { id: 'test-id' },
    });

    const deleteAlert = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<PetList />);

    const onOKPress = jest.spyOn(Alert, 'alert');

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-delete-card-icon'), 'onClick');
    });
    expect(deleteAlert).toHaveBeenCalled();

    const onOK = deleteAlert.mock.calls[0]?.[2]?.[1];
    onOK?.onPress();

    expect(onOKPress).toHaveBeenCalled();
  });

  test('deletePetById throws an error that is an instance of ERROR ', async () => {
    require('@/lib/api').getPetList.mockReturnValue({
      data: Object.values(petObj),
      error: null,
    });

    // deletePetById catch and ERROR is instance of ERROR
    require('@/lib/api').deletePetById.mockRejectedValue(new Error('test error message'));

    require('@/store/store').usePetStore.mockReturnValue({
      setPetList: jest.fn(),
      pets: Object.values(petObj),
      removePetFromStore: jest.fn(),
    });

    require('@/store/store').useToastStore.mockReturnValue({
      setToast: jest.fn(),
    });

    require('@/components/context/UserContext').useUser.mockReturnValue({
      session: { user: 'user-test1' },
    });

    const deleteAlert = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<PetList />);

    const onOKPress = jest.spyOn(Alert, 'alert');

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-delete-card-icon'), 'onClick');
    });
    expect(deleteAlert).toHaveBeenCalled();

    const onOK = deleteAlert.mock.calls[0]?.[2]?.[1];
    onOK?.onPress();

    expect(onOKPress).toHaveBeenCalled();
  });
});
