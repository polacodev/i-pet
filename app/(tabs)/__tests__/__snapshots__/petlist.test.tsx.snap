// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetList /> PetList Data empty with User Session renders correctly and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#0891b2",
            "fontSize": 30,
          },
          [
            undefined,
          ],
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="pet-icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          undefined,
        ]
      }
      testID="pet-title"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "bottom": 30,
          "opacity": 1,
          "position": "absolute",
          "right": 50,
        }
      }
      testID="pet-button-register"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#0891b2",
              "fontSize": 60,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<PetList /> PetList with No User Session renders correctly and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          undefined,
        ]
      }
      testID="pet-title"
    >
      pet_list_message_2
    </Text>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          undefined,
        ]
      }
      testID="pet-title"
    >
      pet_list_message_3
    </Text>
  </View>
</View>
`;

exports[`<PetList /> goToPetRegister is called with User Session and PetList and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <RCTScrollView
      data={
        [
          {
            "id": "testId",
            "inserted_at": "testInsertedId",
            "pet_age": "testPetAge",
            "pet_breed": "testPetBreed",
            "pet_gender": "testPetGender",
            "pet_image": "https://fakeurl/fakeimage/test.png",
            "pet_medical_condition": "testPetMC",
            "pet_name": "testPetName",
            "pet_type": "testPetType",
            "profiles": {
              "country_code": "testProfileCountryCode",
              "email": "testProfileEmail",
              "id": "testProfileId",
              "phone": "testProfilePhone",
            },
            "user_id": "testUserId",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      style={
        {
          "width": "100%",
        }
      }
      testID="pet-list"
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="pet-details-card"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#f6f6f6",
                  },
                  {
                    "borderColor": "#d1d1d1",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "gap": 5,
                    "marginHorizontal": 10,
                    "marginVertical": 5,
                    "paddingBottom": 10,
                    "paddingTop": 4,
                  },
                ]
              }
              testID="pet-view"
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "#f6f6f6",
                    },
                    {
                      "alignItems": "flex-end",
                      "marginHorizontal": 10,
                    },
                  ]
                }
                testID="pet-view"
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "flex-end",
                      "justifyContent": "flex-end",
                      "opacity": 1,
                      "width": "15%",
                    }
                  }
                  testID="pet-delete-card-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#6b7280",
                          "fontSize": 20,
                        },
                        [
                          undefined,
                        ],
                        {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="pet-icon"
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "backgroundColor": "#f6f6f6",
                    },
                    {
                      "alignItems": "center",
                      "marginHorizontal": 10,
                    },
                  ]
                }
                testID="pet-view"
              >
                <Image
                  resizeMode="cover"
                  source={
                    {
                      "uri": "https://fakeurl/fakeimage/test.png",
                    }
                  }
                  style={
                    {
                      "borderRadius": 50,
                      "height": 70,
                      "width": 70,
                    }
                  }
                />
              </View>
              <View
                style={
                  [
                    {
                      "backgroundColor": "#f6f6f6",
                    },
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                      "marginHorizontal": 10,
                    },
                  ]
                }
                testID="pet-view"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 14,
                        "lineHeight": 20,
                      },
                      {
                        "fontWeight": "500",
                      },
                    ]
                  }
                  testID="pet-text"
                >
                  testPetName
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 14,
                        "lineHeight": 20,
                      },
                      {
                        "color": "#6b7280",
                      },
                    ]
                  }
                  testID="pet-text"
                >
                  testPetBreed
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 14,
                        "lineHeight": 20,
                      },
                      {
                        "color": "#6b7280",
                      },
                    ]
                  }
                  testID="pet-text"
                >
                  testPetAge
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "bottom": 30,
          "opacity": 1,
          "position": "absolute",
          "right": 50,
        }
      }
      testID="pet-button-register-with-empty-data"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#0891b2",
              "fontSize": 60,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<PetList /> setLoading true makes appear the ActivityIndicator and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID="pet-view"
>
  <ActivityIndicator />
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
    testID="pet-text"
  />
</View>
`;
